pkg/volume/fc/attacher.go

```diff
@@ -172,12 +172,25 @@ func (detacher *fcDetacher) UnmountDevice(deviceMountPath string) error {
 	if devName == "" {
 		return nil
 	}
+
```

Comment:
So if device is unmounted `devName` will be empty? and then the only guarantee that exists is - polling below we have added to detach/remove device from the node? 

Is there a risk of letting rest in letting rest of the function being executed if `devName` is empty string?

Reply:
If `devName` is an empty string, then the rest of this function (namely `detacher.manager.DetachDisk`) will fail and `UnmountDevice` will be retried indefinitely. Without the device name it does not make sense to even try detaching the device.

The code below is to make sure that the first `UnmountDevice` really detaches a FC volume and second attempt (with empty `devName`) is not necessary.